packages/data-broker/src/services/DataServiceResolver.js

```diff
@@ -94,50 +66,25 @@ export class DataServiceResolver {
    * @param {Entity[]} [orgUnits]
    * @returns {Promise<DataServiceMappingEntry[]>}
    */
-  async resolveDataElements(dataElements, orgUnits = []) {
-    const countryCodes = orgUnits
-      .map(orgUnit => orgUnit.country_code)
-      .filter(countryCode => countryCode !== null && countryCode !== undefined);
+  async resolveDataElements(dataElements, orgUnit) {
+    const countryCode = orgUnit ? orgUnit.country_code : null;
     const dataElementCodes = dataElements.map(de => de.code);
 
-    const mappings =
-      countryCodes.length > 0
-        ? await this.models.dataElementDataService.find({
-            country_code: countryCodes,
-            data_element_code: dataElementCodes,
-          })
-        : [];
+    const mappings = countryCode
+      ? await this.models.dataElementDataService.find({
+          country_code: countryCode,
+          data_element_code: dataElementCodes,
+        })
+      : [];
 
     const resolved = [];
     for (const dataElement of dataElements) {
       const deMappings = mappings.filter(m => m.data_element_code === dataElement.code);
 
       if (deMappings.length > 1) {
-        // If they all resolve to the same data service, this is fine
-        const resolveToTheSameDataService = (mappingA, mappingB) => {
-          if (mappingA.service_type !== mappingB.service_type) return false;
-          return isequal(mappingA.service_config, mappingB.service_config);
-        };
-        for (const mappingA of deMappings) {
-          for (const mappingB of deMappings) {
-            if (mappingA === mappingB) continue;
-            if (!resolveToTheSameDataService(mappingA, mappingB)) {
-              throw new Error(
-                `Conflicting mappings found for Data Source ${
-                  dataElement.code
-                } when fetching for orgUnits ${orgUnits.map(o => o.code).join(',')}`,
-              );
-            }
-          }
-        }
-        // all data element mappings are the same, use first
-        const [deMapping] = deMappings;
-        resolved.push({
-          dataSource: dataElement,
-          service_type: deMapping.service_type,
-          config: deMapping.service_config,
-        });
-        continue;
+        throw new Error(
+          `Multiple mappings found for Data Element ${dataElement.code} for country ${countryCode}`,
```

Comment:
I wonder if this should be a constraint in the database? It seems like we'd never there to be multiple `dataElementDataservices` with the same `country_code` and `data_element_code`. Optional though 👍